source/Halibut.Tests/PollingServiceTimeoutsFixture.cs

```diff
@@ -105,5 +133,46 @@ await using (var clientAndService = await clientAndServiceTestCase.CreateTestCas
                 }, TimeSpan.FromSeconds(30), Logger, CancellationToken);
             }
         }
+
+        [Test]
+        [LatestClientAndLatestServiceTestCases(testNetworkConditions: false, testListening: false)]
+        public async Task WhenThePollingRequestHasBegunTransfer_TheRequestShouldTimeout_AndTheTransferringPendingRequestCancelled(ClientAndServiceTestCase clientAndServiceTestCase)
+        {
+            var halibutTimeoutsAndLimits = new HalibutTimeoutsAndLimitsForTestsBuilder().Build();
+            halibutTimeoutsAndLimits.PollingRequestQueueTimeout = TimeSpan.FromSeconds(5);
+            halibutTimeoutsAndLimits.TcpClientReceiveResponseTimeout = TimeSpan.FromSeconds(6);
+            halibutTimeoutsAndLimits.RelyOnConnectionTimeoutsInsteadOfPollingRequestMaximumMessageProcessingTimeout = true;
+
+            var waitSemaphore = new SemaphoreSlim(0, 1);
+            var connectionsObserver = new TestConnectionsObserver();
+
+            await using (var clientAndService = await clientAndServiceTestCase.CreateTestCaseBuilder()
+                             .As<LatestClientAndLatestServiceBuilder>()
+                             .WithDoSomeActionService(() => waitSemaphore.Wait(CancellationToken))
+                             .WithHalibutTimeoutsAndLimits(halibutTimeoutsAndLimits)
+                             .WithInstantReconnectPollingRetryPolicy()
+                             .WithConnectionObserverOnTcpServer(connectionsObserver)
+                             .Build(CancellationToken))
+            {
+                var doSomeActionClient = clientAndService.CreateAsyncClient<IDoSomeActionService, IAsyncClientDoSomeActionServiceWithOptions>();
+
+                var stopwatch = Stopwatch.StartNew();
+                (await AssertException.Throws<HalibutClientException>(async () => await doSomeActionClient.ActionAsync(new(CancellationToken, CancellationToken.None))))
+                    .And.Message.Should().ContainAny(
+                        "Unable to read data from the transport connection: Connection timed out.",
+                        "Unable to read data from the transport connection: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond");
+                stopwatch.Stop();
+
+                stopwatch.Elapsed.Should().BeLessThan(TimeSpan.FromSeconds(15), "Should have timed out quickly");
+                
+                Wait.UntilActionSucceeds(() =>
+                {
+                    connectionsObserver.ConnectionClosedCount.Should().Be(1, "Cancelling the PendingRequest will time out, causing the connection to be closed");
```

Comment:
Maybe change the msg, we just want to see that when we timeout waiting for the response we kill that connection.